Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/BlendColorGradingLuts.azsl

```diff
@@ -62,40 +63,18 @@ ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
 [[range(0, 4)]]
 option uint o_numSourceLuts = 0;
 
-float3 ShaperToLinear(float3 shaperColor, ShaperType shaperType, float shaperBias, float shaperScale)
-{
-    // Apply the inverse of the shaper function to give the color in the working color space
-    float3 linearColor = shaperColor;
-    if (shaperType == ShaperType::ShaperLinear)
-    {
-        linearColor = (shaperColor - shaperBias)/shaperScale;
-    }
-    else if (shaperType == ShaperType::ShaperLog2)
-    {
-        linearColor = pow(2.0, (shaperColor - shaperBias)/shaperScale);
-    }
-    return linearColor;
-}
-
-float3 LinearToShaper(float3 linearColor, ShaperType shaperType, float shaperBias, float shaperScale)
-{
-    // Convert from working color space to lut coordinates by applying the shaper function
-    float3 shaperColor = linearColor;
-    if (shaperType == ShaperType::ShaperLinear)
-    {
-        shaperColor = linearColor * shaperScale + shaperBias;
-    }
-    else if (shaperType == ShaperType::ShaperLog2)
-    {
-        shaperColor = log2(linearColor) * shaperScale + shaperBias;
-    }
-    return shaperColor;
-}
-
 float3 GetSourceLutLinearColor(float3 baseColor, Texture3D<float4> sourceLut, ShaperType shaperType, float shaperBias, float shaperScale)
 {
     // Convert from reference linearColor to the lutCoordinate for this Lut
     float3 lutCoord = LinearToShaper(baseColor, shaperType, shaperBias, shaperScale);
+
+    // Adjust coordinate to the domain excluding the outer half texel in all directions
+    uint3 outputDimensions;
+    sourceLut.GetDimensions(outputDimensions.x, outputDimensions.y, outputDimensions.z);
+    float3 coordBias = 1.0/(2.0 * outputDimensions);
+    float3 coordScale = (outputDimensions-1.0)/outputDimensions;
```

Comment:
Maybe store `outputDimension - 1.0` and later do `baseCoord = outPixel / outputDimension` for terseness

Reply:
Switching up the baseCoord calculation makes a lot of sense - it condenses down to one line. The outputDimensoins here though are for the source lut, not the blend lut, so I think this is the only place where outputDimensions - 1.0 is used.